### YamlMime:ManagedReference
items:
- uid: System.Runtime.InteropServices.ExternalException
  id: ExternalException
  children:
  - System.Runtime.InteropServices.ExternalException.#ctor
  - System.Runtime.InteropServices.ExternalException.#ctor(System.String)
  - System.Runtime.InteropServices.ExternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Runtime.InteropServices.ExternalException.#ctor(System.String,System.Exception)
  - System.Runtime.InteropServices.ExternalException.#ctor(System.String,System.Int32)
  - System.Runtime.InteropServices.ExternalException.ErrorCode
  - System.Runtime.InteropServices.ExternalException.ToString
  langs:
  - csharp
  name: ExternalException
  nameWithType: ExternalException
  fullName: System.Runtime.InteropServices.ExternalException
  type: Class
  summary: "所有 COM 互操作异常和结构化的异常处理 (SEH) 异常的异常基类型。"
  remarks: "若要增强旧系统和公共语言运行时之间的互操作性<xref:System.Runtime.InteropServices.ExternalException.ErrorCode%2A>属性存储一个整数值 (HRESULT)，标识错误。</xref:System.Runtime.InteropServices.ExternalException.ErrorCode%2A> 用户定义的异常应永远不会派生自`ExternalException`，和`ExternalException`永远不应由用户代码引发。 使用派生自的特定异常`ExternalException`相反。       `ExternalException`使用具有值 0x80004005 HRESULT E_FAIL。       `ExternalException`使用默认<xref:System.Object.Equals%2A>支持引用相等性的实现。</xref:System.Object.Equals%2A>       有关初始的实例值的列表`ExternalException`，请参阅`ExternalException`构造函数。"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class ExternalException : SystemException
  inheritance:
  - System.Exception
  - System.SystemException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Runtime.InteropServices.ExternalException.#ctor
  id: '#ctor'
  parent: System.Runtime.InteropServices.ExternalException
  langs:
  - csharp
  name: ExternalException()
  nameWithType: ExternalException.ExternalException()
  fullName: System.Runtime.InteropServices.ExternalException.ExternalException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Runtime.InteropServices
  summary: "初始化的新实例<xref uid=&quot;langword_csharp_ExternalException&quot; name=&quot;ExternalException&quot; href=&quot;&quot;></xref>使用默认属性的类。"
  remarks: "下表显示的实例的初始属性值`ExternalException`。      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|空引用 (`Nothing`在 Visual Basic 中)。 |  |<xref:System.Exception.Message%2A>|本地化的错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ExternalException ();
    parameters: []
  overload: System.Runtime.InteropServices.ExternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Runtime.InteropServices.ExternalException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Runtime.InteropServices.ExternalException
  langs:
  - csharp
  name: ExternalException(String)
  nameWithType: ExternalException.ExternalException(String)
  fullName: System.Runtime.InteropServices.ExternalException.ExternalException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Runtime.InteropServices
  summary: "初始化的新实例<xref uid=&quot;langword_csharp_ExternalException&quot; name=&quot;ExternalException&quot; href=&quot;&quot;></xref>与指定的错误消息的类。"
  remarks: "下表显示的实例的初始属性值`ExternalException`。      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|空引用 (`Nothing`在 Visual Basic 中)。 |  |<xref:System.Exception.Message%2A>|错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ExternalException (string message);
    parameters:
    - id: message
      type: System.String
      description: "指定异常的原因的错误消息。"
  overload: System.Runtime.InteropServices.ExternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Runtime.InteropServices.ExternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Runtime.InteropServices.ExternalException
  langs:
  - csharp
  name: ExternalException(SerializationInfo,StreamingContext)
  nameWithType: ExternalException.ExternalException(SerializationInfo,StreamingContext)
  fullName: System.Runtime.InteropServices.ExternalException.ExternalException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Runtime.InteropServices
  summary: "初始化的新实例<xref uid=&quot;langword_csharp_ExternalException&quot; name=&quot;ExternalException&quot; href=&quot;&quot;></xref>从序列化数据的类。"
  remarks: "在反序列化，以便重建通过流传输的异常对象过程中调用此构造函数。 有关详细信息，请参阅[XML 和 SOAP 序列化](~/add/includes/ajax-current-ext-md.md)。"
  syntax:
    content: protected ExternalException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "承载序列化的对象数据的对象。"
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "有关源或目标的上下文信息。"
  overload: System.Runtime.InteropServices.ExternalException.#ctor*
  exceptions:
  - type: System.ArgumentNullException
    commentId: T:System.ArgumentNullException
    description: "<code>info</code>is <xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;></xref>."
  platform:
  - net462
- uid: System.Runtime.InteropServices.ExternalException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Runtime.InteropServices.ExternalException
  langs:
  - csharp
  name: ExternalException(String,Exception)
  nameWithType: ExternalException.ExternalException(String,Exception)
  fullName: System.Runtime.InteropServices.ExternalException.ExternalException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Runtime.InteropServices
  summary: "初始化的新实例<xref href=&quot;System.Runtime.InteropServices.ExternalException&quot;></xref>使用指定的错误消息和对导致此异常的内部异常的引用的类。"
  remarks: "前一个异常的直接结果应包含中的前一个异常的引用引发的异常，<xref:System.Exception.InnerException%2A>属性。</xref:System.Exception.InnerException%2A> <xref:System.Exception.InnerException%2A>属性返回相同的值传递到构造函数中，或`null`如果<xref:System.Exception.InnerException%2A>属性不会提供给构造函数的内部异常值。</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       下表显示<xref:System.Runtime.InteropServices.ExternalException>.</xref:System.Runtime.InteropServices.ExternalException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|内部异常引用。 |  |<xref:System.Exception.Message%2A>|错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ExternalException (string message, Exception inner);
    parameters:
    - id: message
      type: System.String
      description: "解释异常原因的错误消息。"
    - id: inner
      type: System.Exception
      description: "导致当前异常的异常。 如果`inner`参数不是<xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;></xref>中, 引发当前异常<xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;></xref>处理内部异常的块。"
  overload: System.Runtime.InteropServices.ExternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Runtime.InteropServices.ExternalException.#ctor(System.String,System.Int32)
  id: '#ctor(System.String,System.Int32)'
  parent: System.Runtime.InteropServices.ExternalException
  langs:
  - csharp
  name: ExternalException(String,Int32)
  nameWithType: ExternalException.ExternalException(String,Int32)
  fullName: System.Runtime.InteropServices.ExternalException.ExternalException(String,Int32)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Runtime.InteropServices
  summary: "初始化的新实例<xref uid=&quot;langword_csharp_ExternalException&quot; name=&quot;ExternalException&quot; href=&quot;&quot;></xref>使用指定的错误消息和错误的 HRESULT 的类。"
  remarks: "下表显示的实例的初始属性值`ExternalException`。      |属性 |值 |  |--------------|-----------|  |<xref:System.Runtime.InteropServices.ExternalException.ErrorCode%2A>|错误的 HRESULT。 |  |<xref:System.Exception.InnerException%2A>|空引用 (`Nothing`在 Visual Basic 中)。 |  |<xref:System.Exception.Message%2A>|错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A> </xref:System.Runtime.InteropServices.ExternalException.ErrorCode%2A>"
  syntax:
    content: public ExternalException (string message, int errorCode);
    parameters:
    - id: message
      type: System.String
      description: "指定异常的原因的错误消息。"
    - id: errorCode
      type: System.Int32
      description: "错误的 HRESULT。"
  overload: System.Runtime.InteropServices.ExternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Runtime.InteropServices.ExternalException.ErrorCode
  id: ErrorCode
  parent: System.Runtime.InteropServices.ExternalException
  langs:
  - csharp
  name: ErrorCode
  nameWithType: ExternalException.ErrorCode
  fullName: System.Runtime.InteropServices.ExternalException.ErrorCode
  type: Property
  assemblies:
  - mscorlib
  namespace: System.Runtime.InteropServices
  summary: "获取<xref uid=&quot;langword_csharp_HRESULT&quot; name=&quot;HRESULT&quot; href=&quot;&quot;></xref>的错误。"
  remarks: "请参阅[常见的 HRESULT 值](https://msdn.microsoft.com/library/windows/desktop/aa378137.aspx)越多的列表经常遇到的错误代码。 `HRESULT`Winerror.h 标头文件中定义值。"
  syntax:
    content: public virtual int ErrorCode { get; }
    return:
      type: System.Int32
      description: "<xref uid=&quot;langword_csharp_HRESULT&quot; name=&quot;HRESULT&quot; href=&quot;&quot;> </xref>的错误。"
  overload: System.Runtime.InteropServices.ExternalException.ErrorCode*
  exceptions: []
  platform:
  - net462
- uid: System.Runtime.InteropServices.ExternalException.ToString
  id: ToString
  parent: System.Runtime.InteropServices.ExternalException
  langs:
  - csharp
  name: ToString()
  nameWithType: ExternalException.ToString()
  fullName: System.Runtime.InteropServices.ExternalException.ToString()
  type: Method
  assemblies:
  - mscorlib
  namespace: System.Runtime.InteropServices
  summary: "返回一个字符串，包含错误的 HRESULT。"
  syntax:
    content: public override string ToString ();
    parameters: []
    return:
      type: System.String
      description: "一个字符串，表示 HRESULT。"
  overload: System.Runtime.InteropServices.ExternalException.ToString*
  exceptions: []
  platform:
  - net462
references:
- uid: System.SystemException
  isExternal: false
  name: System.SystemException
- uid: System.ArgumentNullException
  isExternal: true
  name: System.ArgumentNullException
- uid: System.Runtime.InteropServices.ExternalException.#ctor
  parent: System.Runtime.InteropServices.ExternalException
  isExternal: false
  name: ExternalException()
  nameWithType: ExternalException.ExternalException()
  fullName: System.Runtime.InteropServices.ExternalException.ExternalException()
- uid: System.Runtime.InteropServices.ExternalException.#ctor(System.String)
  parent: System.Runtime.InteropServices.ExternalException
  isExternal: false
  name: ExternalException(String)
  nameWithType: ExternalException.ExternalException(String)
  fullName: System.Runtime.InteropServices.ExternalException.ExternalException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Runtime.InteropServices.ExternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Runtime.InteropServices.ExternalException
  isExternal: false
  name: ExternalException(SerializationInfo,StreamingContext)
  nameWithType: ExternalException.ExternalException(SerializationInfo,StreamingContext)
  fullName: System.Runtime.InteropServices.ExternalException.ExternalException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Runtime.InteropServices.ExternalException.#ctor(System.String,System.Exception)
  parent: System.Runtime.InteropServices.ExternalException
  isExternal: false
  name: ExternalException(String,Exception)
  nameWithType: ExternalException.ExternalException(String,Exception)
  fullName: System.Runtime.InteropServices.ExternalException.ExternalException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Runtime.InteropServices.ExternalException.#ctor(System.String,System.Int32)
  parent: System.Runtime.InteropServices.ExternalException
  isExternal: false
  name: ExternalException(String,Int32)
  nameWithType: ExternalException.ExternalException(String,Int32)
  fullName: System.Runtime.InteropServices.ExternalException.ExternalException(String,Int32)
- uid: System.Int32
  parent: System
  isExternal: true
  name: Int32
  nameWithType: Int32
  fullName: System.Int32
- uid: System.Runtime.InteropServices.ExternalException.ErrorCode
  parent: System.Runtime.InteropServices.ExternalException
  isExternal: false
  name: ErrorCode
  nameWithType: ExternalException.ErrorCode
  fullName: System.Runtime.InteropServices.ExternalException.ErrorCode
- uid: System.Runtime.InteropServices.ExternalException.ToString
  parent: System.Runtime.InteropServices.ExternalException
  isExternal: false
  name: ToString()
  nameWithType: ExternalException.ToString()
  fullName: System.Runtime.InteropServices.ExternalException.ToString()
- uid: System.Runtime.InteropServices.ExternalException.#ctor*
  parent: System.Runtime.InteropServices.ExternalException
  isExternal: false
  name: ExternalException
  nameWithType: ExternalException.ExternalException
- uid: System.Runtime.InteropServices.ExternalException.ErrorCode*
  parent: System.Runtime.InteropServices.ExternalException
  isExternal: false
  name: ErrorCode
  nameWithType: ExternalException.ErrorCode
- uid: System.Runtime.InteropServices.ExternalException.ToString*
  parent: System.Runtime.InteropServices.ExternalException
  isExternal: false
  name: ToString
  nameWithType: ExternalException.ToString
